commons/zenoh-runtime/src/lib.rs

```diff
@@ -141,6 +158,9 @@ impl ZRuntime {
                 }
             }
         }
+
+        let f = tracing::Instrument::instrument(f, tracing::Span::current());
```

Comment:
For performance reasons I'd put all these tracing calls behind a feature flag that is disabled by default.

Reply:
Makes sense.

I'll add it in.